code/modules/mob/living/living_defense.dm

```diff
@@ -413,10 +413,27 @@
 
 	if(!damage || !istype(user))
 		return
-	if(damagetype == BRUTE)
-		adjustBruteLoss(damage)
-	else
-		adjustFireLoss(damage)
+	
+	var/used_divisor = 1
+	if(isliving(user))
+		var/mob/living/L = user
+		used_divisor = L.armor_divisor
+	var/attack_BP = BP_CHEST
+	if(prob(20))
+		switch(rand(1,6))
+			if(1)
+				attack_BP = BP_L_LEG
+			if(2)
+				attack_BP = BP_R_LEG
+			if(3)
+				attack_BP = BP_R_ARM
+			if(4)
+				attack_BP = BP_L_ARM
+			if(5)
+				attack_BP = BP_GROIN
+			if(6)
+				attack_BP = BP_HEAD
```

Comment:
```suggestion
		attack_BP = pick(list(BP_L_LEG, BP_R_LEG, BP_R_ARM, BP_L_ARM, BP_GROIN, BP_HEAD))
```
I will try to see if one such list already exists somewhere else. This is a more preferred way to do this. Plus this does not guarantee if the picked limb exists (such as being destroyed in prior attacks or just straight up missing.)

Reply:
this is in fact how projectiles do it, and i think the occasional 'mob missing because it went for a limb you don't have' doesn't compensate for missing an entire limb

Reply:
![image](https://github.com/user-attachments/assets/e3bc859b-dd9f-438b-b4c8-8d2dea2bd9c4)

You mean this process is used for both melee and ranged attacks?

Anyway, my process of though was following: `adjustBruteLoss` calls `take_overall_damage` for human mobs. See here:

https://github.com/Equinox-SS13/equinox-sojourn/blob/5c8413cdc3edb13d5b2f551bc6cc222e386fa08b/code/modules/mob/living/carbon/human/human_damage.dm#L280-L301

While this code ignores armor since it deals the damage directly to limbs and such. It distributes the damage across several limbs while they still can take damage, if there is a remainder after dealing damage to a limb, it picks another and repeats until there are no more "damageable" limbs left to damage or destroy.

But the code seems to be fine as-is for now, I suppose. But I feel like I would have preferred the randomness to be equally uniform.